src/window.c

```diff
@@ -157,6 +157,37 @@ log_frame_layout(frame_T *frame)
 }
 #endif
 
+/*
+ * Check if the current window is allowed to move to a different buffer.
+ * If the window has 'winfixbuf', this function will return FALSE.
+ */
+int check_can_set_curbuf_disabled(void)
+{
+    if (curwin->w_p_wfb)
+    {
+	semsg("%s", e_winfixbuf_cannot_go_to_buffer);
+	return FALSE;
+    }
+
+    return TRUE;
+}
+
+/*
+ * Check if the current window is allowed to move to a different buffer.
+ * If the window has 'winfixbuf', then forceit must be TRUE or this function
+ * will return FALSE.
+ */
+int check_can_set_curbuf_forceit(int forceit)
+{
+    if (!forceit && curwin->w_p_wfb)
+    {
+	semsg("%s", e_winfixbuf_cannot_go_to_buffer);
+	return FALSE;
```

Comment:
and here please

Reply:
A lot of tests should be using this block of code already

<details>
<summary>All tests</summary>

```
Test_Next
Test_argedit
Test_argglobal
Test_arglocal
Test_args
Test_bNext
Test_bfirst
Test_blast
Test_bmodified
Test_bnext
Test_bprevious
Test_brewind
Test_buffer
Test_cbuffer
Test_cexpr
Test_cfile
Test_djump
Test_drop
Test_edit
Test_enew
Test_ex
Test_find
Test_first
Test_grep
Test_ijump
Test_last
Test_ltag
Test_marks_mappings_fail
Test_next
Test_next_same_buffer
Test_normal_ctrl_o_fail
Test_normal_ctrl_rightmouse
Test_normal_ctrl_square_bracket_right
Test_normal_ctrl_t
Test_normal_ctrl_w_ctrl_square_bracket_right
Test_normal_ctrl_w_g_ctrl_square_bracket_right
Test_normal_g_ctrl_square_bracket_right
Test_normal_g_rightmouse
Test_normal_g_square_bracket_right
Test_normal_gt
Test_normal_square_bracket_left_ctrl_d
Test_normal_square_bracket_left_ctrl_i
Test_normal_square_bracket_right_ctrl_d
Test_normal_square_bracket_right_ctrl_i
Test_normal_v_ctrl_square_bracket_right
Test_normal_v_g_ctrl_square_bracket_right
Test_pop
Test_previous
Test_rewind
Test_short_option
Test_tNext
Test_tag
Test_tfirst
Test_tjump
Test_tlast
Test_tnext
Test_tprevious
Test_view
Test_vimgrep
Test_vimgrepadd
Test_visual
Test_wNext
Test_windo
Test_wnext
Test_wprevious
```
</details>